packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/common/bboxUtils.ts

```diff
@@ -0,0 +1,76 @@
+import { Bbox, TextSpan } from '../../types';
+import { spanIntersection, spanLen } from './textSpanUtils';
+
+export const LEFT = 0;
+export const TOP = 1;
+export const RIGHT = 2;
+export const BOTTOM = 3;
+
+/**
+ * Check whether two bbox intersect
+ *
+ * Same to `intersects` in DocumentPreview/utils/box.ts,
+ * but for type `Bbox`, which doesn't have page property
+ * @param boxA one bbox
+ * @param boxB another bbox
+ * @returns true iff boxA and boxB are overwrapped
+ */
+export function bboxIntersects(boxA: Bbox, boxB: Bbox) {
+  const [leftA, topA, rightA, bottomA] = boxA;
+  const [leftB, topB, rightB, bottomB] = boxB;
+  return !(leftB >= rightA || rightB <= leftA || topB >= bottomA || bottomB <= topA);
+}
+
+/**
+ * Get bbox for a text span assuming each character takes horizontal spaces evenly
```

Comment:
> What are the implications here if the characters are not evenly spaced?

In the case, left and right ends of highlight box don't exactly match to the beginning/ending characters of highlight span.

> And how often will we run into this scenario?

Currently, OCR text will be run into this case because the text won't be available as text object in PDF. I'm planning to improve it by implementing a logic to find bbox using DOM later.